six/modules/c++/six/source/NITFReadControl.cpp

```diff
@@ -178,80 +269,39 @@ void NITFReadControl::load(nitf::IOInterface& ioInterface,
 
     for (size_t i = 0, productNum = 0; desIter != des.end(); ++desIter, ++i)
     {
-        // Get a segment ref
-        nitf::DESegment segment = (nitf::DESegment) * desIter;
+        nitf::DESegment segment = (nitf::DESegment) *desIter;
         nitf::DESubheader subheader = segment.getSubheader();
-        std::string desid = subheader.getTypeID().toString();
-        str::trim(desid);
-
-        // SICD/SIDD 1.0 specify DESID as XML_DATA_CONTENT
-        // Older versions of the spec specified it as SICD_XML/SIDD_XML
-        // Here we'll accept any of these under the assumption that it's not
-        // such an old version of the spec that the XML layout itself has
-        // changed (if it did, XMLControl will end up throwing anyway)
-        if (desid == "XML_DATA_CONTENT" ||
-            desid == "SICD_XML" ||
-            desid == "SIDD_XML")
+        if (getDataType(segment) == DataType::NOT_SET)
```

Comment:
Look at the SIDD 1.1 branch... the FTITLE check on line 254ish needs to get changed over to this too